### PR TITLE
[feat]: [CDS-49534]: Change handler for 'tags_info_ng' table

### PR DESCRIPTION
**Pull request by Abhishek-CDC**
## Describe your changes
Added change handler for 'tags_info_ng' table to sync tags info for different entities
## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/42083)

### CODE REVIEW

**Review by abhinavmittal-harness — DISMISSED** [review]
:lgtm:

> Reviewed 10 of 11 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42083#-:-NKvQVXk5Yvn0THr47mE:bnfp4nl)* status: Some of the checks are too old. 11 file(s) to review. 1 of 2 approvals obtained.

**Review by vaibhav-si — CHANGES_REQUESTED** [review]
There is a new framework for data sync..Debezium Service that PIP Team is working on. Shouldn't we use that going forward? cc @naidu-harness @shalini-agr 

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42083#-:-NKvpmQo12H6xAdnMPNA:bt4mwrq)* status: Some of the checks are too old. 11 file(s) to review. 1 change requests. 1 of 2 approvals obtained. 

___
*[`110-change-data-capture/src/main/java/io/harness/changehandlers/TagsInfoCDChangeDataHandlerHelper.java` line 84 at r1](https://reviewable.io/reviews/harness/harness-core/42083#-NKvo160BQ_KNtEfyEzY:-NKvo160BQ_KNtEfyEzZ:bphigql) ([raw file](https://github.com/harness/harness-core/blob/0e12ff532e06daa6c4405eb8ad40f6a028e95061/110-change-data-capture/src/main/java/io/harness/changehandlers/TagsInfoCDChangeDataHandlerHelper.java#L84)):*
> ```Java
>   }
> 
>   public String getParentType(ChangeEvent<?> changeEvent) {
> ```

what does parent mean? please add comments
___
*[`110-change-data-capture/src/main/java/io/harness/changehandlers/TagsInfoCDChangeDataHandlerHelper.java` line 118 at r1](https://reviewable.io/reviews/harness/harness-core/42083#-NKvoQff14gk4u1lV1gH:-NKvoQfgAPly0VC8IEQU:b-r0zsrk) ([raw file](https://github.com/harness/harness-core/blob/0e12ff532e06daa6c4405eb8ad40f6a028e95061/110-change-data-capture/src/main/java/io/harness/changehandlers/TagsInfoCDChangeDataHandlerHelper.java#L118)):*
> ```Java
>     return null;
>   }
> ```

add comments explaining the input and output format as an example
___
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/PipelineEntity.java` line 75 at r1](https://reviewable.io/reviews/harness/harness-core/42083#-NKvpQyS2mUXmYv2Q6YK:-NKvpQyS2mUXmYv2Q6YL:b23olvl) ([raw file](https://github.com/harness/harness-core/blob/0e12ff532e06daa6c4405eb8ad40f6a028e95061/pipeline-service/service/src/main/java/io/harness/pms/pipeline/PipelineEntity.java#L75)):*
> ```Java
> @HarnessEntity(exportable = true)
> @ChangeDataCapture(table = "tags_info", dataStore = "pms-harness", fields = {}, handler = "TagsInfoCD")
> @ChangeDataCapture(table = "tags_info_ng", dataStore = "ng-harness", fields = {}, handler = "TagsInfoNGCD")
> ```

what's the diff. in tags_info and tags_info_ng?
And datastore is pms-harness  for tags_info but ng-harness for tags_info_ng..is this expected?

**Comment by Abhishek-CDC** [comment]
*[`110-change-data-capture/src/main/java/io/harness/changehandlers/TagsInfoCDChangeDataHandlerHelper.java` line 84 at r1](https://reviewable.io/reviews/harness/harness-core/42083#-NKvo160BQ_KNtEfyEzY:-NKwX0hLCVOypMkSWhlv:b51t5so) ([raw file](https://github.com/harness/harness-core/blob/0e12ff532e06daa6c4405eb8ad40f6a028e95061/110-change-data-capture/src/main/java/io/harness/changehandlers/TagsInfoCDChangeDataHandlerHelper.java#L84)):*
<details><summary><i>Previously, vaibhav-si (Vaibhav Singhal) wrote…</i></summary><blockquote>

what does parent mean? please add comments
</blockquote></details>

Parent we are referring to entity for which we are saving tags. added comments for the same.

**Comment by Abhishek-CDC** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/PipelineEntity.java` line 75 at r1](https://reviewable.io/reviews/harness/harness-core/42083#-NKvpQyS2mUXmYv2Q6YK:-NKwXKBc0AvxbHSv_eeF:bvaw2c6) ([raw file](https://github.com/harness/harness-core/blob/0e12ff532e06daa6c4405eb8ad40f6a028e95061/pipeline-service/service/src/main/java/io/harness/pms/pipeline/PipelineEntity.java#L75)):*
<details><summary><i>Previously, vaibhav-si (Vaibhav Singhal) wrote…</i></summary><blockquote>

what's the diff. in tags_info and tags_info_ng?
And datastore is pms-harness  for tags_info but ng-harness for tags_info_ng..is this expected?
</blockquote></details>

tags_info had some issues with design, to address them we have created tags_info_ng table with more details. made changes to use pms-harness as datastore.

**Comment by Abhishek-CDC** [comment]
Will check for future implementations, thanks for suggestion.

File: 110-change-data-capture/src/main/java/io/harness/changehandlers/TagsInfoNGCDChangeDataHandler.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+      columnValueMapping.remove("moduleinfo_is_private");
+      columnValueMapping.remove("pr");
+    }
+
+    Map<String, String> keyMap = new HashMap<>();
+
+    if (changeEvent != null) {
+      keyMap.put(PARENT_TYPE, TagsInfoCDChangeDataHandlerHelper.getParentType(changeEvent));
+      keyMap.put(ID, changeEvent.getUuid());
+    }
```
Comment (yogesh-chauhan) [inline 110-change-data-capture/src/main/java/io/harness/changehandlers/TagsInfoNGCDChangeDataHandler.java:54]: if keyMap is empty or changeEvent is null, can we return here itself ?

**Review by yogesh-chauhan — CHANGES_REQUESTED** [review]

**Comment by Abhishek-CDC** [comment]
*[`110-change-data-capture/src/main/java/io/harness/changehandlers/TagsInfoNGCDChangeDataHandler.java` line 54 at r2](https://reviewable.io/reviews/harness/harness-core/42083#-NKSuRVuo69tamzqRlh7-r2-54:-NL-h5CS2Tl9BOpPDAsT:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/ff25334e55745a2c26e567a561e926e0d1d6c476/110-change-data-capture/src/main/java/io/harness/changehandlers/TagsInfoNGCDChangeDataHandler.java#L54)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

if keyMap is empty or changeEvent is null, can we return here itself ?
</blockquote></details>

Done

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 5 of 11 files at r1, 1 of 2 files at r2, 5 of 5 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42083)* status: 6 file(s) to review. 1 change requests. 3. unresolved discussions. 1 of 2 approvals obtained.

**Review by vaibhav-si — APPROVED** [review]
> Reviewed 5 of 11 files at r1, 1 of 2 files at r2, 5 of 5 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42083)* status: :shipit: complete! ✓

**Pull request merged**